clients/template/template.go

```diff
@@ -0,0 +1,119 @@
+package template
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/template"
+)
+
+type Client struct {
+	clients.CLI
+	api.TemplatesApi
+	api.OrganizationsApi
+}
+
+type SummarizeParams struct {
+	OrgId   string
+	OrgName string
+
+	Sources []template.Source
+
+	RenderTableColors  bool
+	RenderTableBorders bool
+}
+
+func (c Client) Summarize(ctx context.Context, params *SummarizeParams) error {
+	if params.OrgId == "" && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+	orgID := params.OrgId
+	if orgID == "" {
+		orgName := params.OrgName
+		if orgName == "" {
+			orgName = c.ActiveConfig.Org
+		}
+		orgs, err := c.GetOrgs(ctx).Org(orgName).Execute()
+		if err != nil {
+			return fmt.Errorf("failed to get ID for org %q: %w", orgName, err)
+		}
+		if len(orgs.GetOrgs()) == 0 {
+			return fmt.Errorf("no orgs found with name %q: %w", orgName, err)
+		}
+		orgID = orgs.GetOrgs()[0].GetId()
+	}
+
+	templates, err := template.ReadSources(ctx, params.Sources)
+	if err != nil {
+		return err
+	}
+
+	// Execute a dry-run to make the server summarize the template(s).
+	req := api.TemplateApply{
+		DryRun:    true,
+		OrgID:     orgID,
+		Templates: templates,
+	}
+	res, err := c.ApplyTemplate(ctx).TemplateApply(req).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to summarize template: %w", err)
```

Comment:
On line 113 below, there's some additional processing in the `Validate` command that seems to still be getting a summary even if an error is returned. If that's what is actually happening, would it be useful to duplicate that kind of logic here? I haven't tried the command out myself, but I'm imagining some kind of useful summary info with maybe where the errors are - maybe it's nothing at all like that though and so it wouldn't be useful.

Reply:
Tried this out, but looks like the server returns an empty summary when you dry-run something with validation errors.

Reply:
influxdata/influxdb#21781